### DISCUSSION TITLE
MEKE tuning

### DISCUSSION THREAD

**Discussion by gustavo-marques** [NCAR/omwg_dev#30] [General]
I want to start a discussion on the MEKE tuning. 

The purpose of run [007](https://github.com/NCAR/omwg_dev/issues/11) is to check the impact of removing `MEKE_EQUILIBRIUM_RESTORING=True` and allowing MEKE to evolve prognostically, as well as setting RESOLN_SCALED_KHTR=True (we expect a minor impact from this change). Here is a list of the MOM parameter changes:

```
!!! changes from Ian !!!!!
MEKE_EQUILIBRIUM_RESTORING = False
MEKE_CB = 100.0
MEKE_CDRAG = 0.0025
MEKE_MIN_GAMMA2 = 1.0E-5
MEKE_VISCOSITY_COEFF_KU = 0.0
MEKE_GMCOEFF = 1.0
MEKE_FRCOEFF = 0.3
MEKE_KHMEKE_FAC = 0.75
RESOLN_SCALED_KHTR = True
KH_RES_SCALE_COEF = 0.9
KH_RES_FN_POWER = 6
USE_SIMPLER_EADY_GROWTH_RATE = True
```

[Here](https://webext.cgd.ucar.edu/MOM/007_vs_015_MEKE/html/intro.html) are the ocean diagnostics comparing [007](https://github.com/NCAR/omwg_dev/issues/11) against [015](https://github.com/NCAR/omwg_dev/issues/14)(baseline). The effect of applying the above changes is small. Are we ready to incorporate these changes into our baselines? 

@iangrooms, @fobryan3

PS: MEKE has a bug that is currently being fixed by Wenda. We can evaluate the effect of this bug once the fix is in the main MOM6 branch.

**Comment by iangrooms** [comment]
It is important to understand both the MEKE prognostic equation and how MEKE is initialized.

## MEKE Initialization
MEKE is initialized in [this](https://github.com/NCAR/MOM6/blob/db64408fd9f403325aa5a006c75bc99b869261ea/src/parameterizations/lateral/MOM_MEKE.F90#L369C12-L369C28) call to [this](https://github.com/NCAR/MOM6/blob/db64408fd9f403325aa5a006c75bc99b869261ea/src/parameterizations/lateral/MOM_MEKE.F90#L753) subroutine.

If `MEKE_EQUILIBRIUM_ALT` is TRUE then MEKE is initialized to a balance between GM conversion and bottom drag (see [this](https://github.com/NCAR/MOM6/blob/db64408fd9f403325aa5a006c75bc99b869261ea/src/parameterizations/lateral/MOM_MEKE.F90#L803) line). It's important to note that this has nothing to do with how the MEKE evolution equation is configured. The balance is between GEOMETRIC GM and quadratic drag regardless of whether or not you're using GEOMETRIC, and the GM production term is not the same as the `GMWork` term computed in `MOM_thickness_diffuse.F90`.

If `MOM_EQUILIBRIUM_ALT` is FALSE then MEKE is initialized by solving for the value of MEKE that balances the GM source and the bottom-drag sink in the MEKE evolution equation. _Except_ that it also ignores how the MEKE evolution equation is configured: It uses a mixing-length scaling for the GM coefficient even if GM is using GEOMETRIC (or something else), and the GM production term is not the same as the `GMWork` term computed in `MOM_thickness_diffuse.F90`.

So in general don't expect MEKE initialization to be an equilibrium of the MEKE evolution equation: There is a transient evolution away from this initial value. In my experience the MEKE initial value is usually larger than the value it eventually reaches, and it takes about 3 or 4 years to equilibrate. (Of course when equilibrium restoring is turned on, there isn't a transient because you're restoring towards the initial value.) Also note that the `MEKE_EQUILIBRIUM_ALT` value is still relevant for the initialization of MEKE even if you turn off MEKE equilibrium restoring. I think historically the 'equilibrium' was first introduced to initialize MEKE, and then later on the option was added to prognostically restore MEKE towards this 'equilibrium' value.

## MEKE Equation Parameters
First note that MEKE 'equilibrium restoring' restores/nudges MEKE towards the values described above. These values are not actually equilibria of the MEKE prognostic equation though, so if you turn MEKE equilibrium restoring on it can (and does) lead to big changes; typically MEKE drops from the initial value.

If we just turn off MEKE equilibrium restoring, MEKE drops, as noted above. Ceterus paribus this means the GM & Redi coefficients drop, which introduces biases. In order to get a configuration with MEKE equilibrium restoring turned off, but with MEKE values (and therefore GM & Redi values) comparable to our old ones, we need to change parameters in such a way that MEKE increases. That means increasing the sources of MEKE and/or reducing the sinks. 

We've already got the GM source at 100%, but there are two options for how to compute the GM source term in MEKE. Setting `USE_SIMPLER_EADY_GROWTH_RATE = TRUE` changes the GM source term (indirectly) [here](https://github.com/NCAR/MOM6/blob/db64408fd9f403325aa5a006c75bc99b869261ea/src/parameterizations/lateral/MOM_lateral_mixing_coeffs.F90#L491). This is the recommended setting, and in my experiments it led to increased MEKE.

Setting `USE_SIMPLER_EADY_GROWTH_RATE = TRUE` is not enough to get MEKE back to the same levels we were seeing with equilibrium restoring turned on. Fortunately there is another MEKE source: Work done on the resolved flow by lateral viscosity can be set as a source for MEKE. That's why `MEKE_FRCOEFF` is positive in the above configuration. Unfortunately it seems that this source term has a bug that is being fixed, but preliminary results suggest that once the bug is fixed we can just re-tune `MEKE_FRCOEFF` to get approximately the same results. I expect we will have to increase `MEKE_FRCOEFF` once the bug is fixed.

Having `MEKE_FRCOEFF` positive was not, by itself, enough to return MEKE to the same level. Increasing above the value of 0.3 set above does increase MEKE, but it produces a different spatial configuration from what we had previously, so I explored other options to increase MEKE. The values
```
MEKE_CB = 100.0
MEKE_CDRAG = 0.0025
MEKE_MIN_GAMMA2 = 1.0E-5
```
all decrease the rate at which MEKE is removed through bottom friction.

These changes are still not enough to increase MEKE back to the levels we saw previously, so to compensate for that I also increased the GEOMETRIC alpha coefficient. We had been using 0.06-0.07, which is below the prediction of 0.2 from Scott's 2017 paper. Increasing the GEOMETRIC alpha to 0.09 increased GM coefficients and GM MOC back to approximately where they were when we were using MEKE equilibrium restoring.

## MEKE Misc
[This](https://doi.org/10.1103/PhysRevFluids.2.113801) paper finds that EKE transport is approximately diffusive, and that the diffusivity is 75% of the tracer diffusivity. So I set `MEKE_KHMEKE_FAC = 0.75`.

The source term from `MEKE_FRCOEFF` is very large in the Maritime continent and along western boundaries in the tropics (at least initially). That leads to very large MEKE in those places. Having a Laplacian viscosity depending on MEKE leads to very large lateral viscosity in those places. I set `MEKE_VISCOSITY_COEFF_KU = 0.0` to avoid that.

**Reply by klindsay28** [reply]
Thanks for the explanations @iangrooms.
However, case [007](https://github.com/NCAR/omwg_dev/issues/11) doesn't appear to have all of the parameter changes that you discuss.
In particular, [007](https://github.com/NCAR/omwg_dev/issues/11) has `MEKE_GEOMETRIC_ALPHA = 0.07`, not `0.09` as you describe.
I'm not certain what value I should use in MOM6-MARBL runs that I'm about to start.

**Reply by iangrooms** [reply]
With 0.07 I expect GM MOC in the SO to be about -9Sv when it should be closer to -12 or -13. I recommend using 0.09. It’s worth checking the GM MOC for run 007 just in case, but I won’t be able until tomorrow.

**Reply by iangrooms** [reply]
I just checked run 007 and indeed the GM MOC in the Southern Ocean is about -9.6Sv, which is a bit low.

**Reply by klindsay28** [reply]
Thanks for taking a look @iangrooms. I started a coupled of runs using `MEKE_GEOMETRIC_ALPHA = 0.09`.
